src/plugins/embeddable/public/plugin.tsx

```diff
@@ -167,11 +188,99 @@ export class EmbeddablePublicPlugin implements Plugin<EmbeddableSetup, Embeddabl
       },
       EmbeddablePanel: getEmbeddablePanelHoc(),
       getEmbeddablePanel: getEmbeddablePanelHoc,
+      migrate: this.migrateEmbeddableInput,
+      extract: this.extract,
+      inject: this.inject,
     };
   }
 
   public stop() {}
 
+  private migrateEmbeddableInput = (input: Serializable) => {
+    const enhancements: Record<string, any> = (input as any).enhanecemnts || {};
+    const factory = this.getEmbeddableFactory((input as SerializableRecord).id as string);
```

Comment:
input.id != type - this won't work.

I think the migrate function will have to ask for type, separate from input.